### ISSUE TITLE
Blink-Led example doesn't build on Nucleo_L476RG, STM32F4_DISCOVERY, Nucleo_F302R8, Nucleo_F401RE 

### ISSUE THREAD

**Issue by giellamoswhard** [zephyrproject-rtos/zephyr#22091]
**Describe the bug**
I've tried to build the BlinkLed example with the boards o_L476RG, STM32F4_DISCOVERY, Nucleo_F302R8, Nucleo_F401RE but it doesn't build 
#error "Choose supported PWM driver" 
**To Reproduce**
1. cd zephyrproject/zephyr/samples/basic/blink_led
2. west build -b nucleo_f401re -- -DBOARD_ROOT=/home/giellamo/projects/zephyrproject/zephyr/boards
3. ../src/main.c:28:2: error: #error "Choose supported PWM driver"
 #error "Choose supported PWM driver"
  ^~~~~
../src/main.c: In function 'main':
../src/main.c:46:31: error: 'PWM_DRIVER' undeclared (first use in this function); did you mean 'PWR_CR_VOS'?
  pwm_dev = device_get_binding(PWM_DRIVER);
                               ^~~~~~~~~~
                               PWR_CR_VOS

**Expected behavior**
It should build

**Environment (please complete the following information):**
 - OS: .g. Linux, MacOS, Windows)
 - Toolchain Zephyr SDK
 - Commit 2.1.99

**Additional context**
The boards seems to miss the pwmleds in the dts

**Comment by erwango** [comment]
Boards supporting/enabling PWM should declare the following or equivalent:
```
	pwmleds {
		compatible = "pwm-leds";		red_pwm_led: red_pwm_led {
			pwms = <&{/soc/timers@40014000/pwm} 1 4>;
		};
	};
	aliases {
        ...
		pwm-led0 = &red_pwm_led;
		...
	};
```

**Comment by sercantun** [comment]
Could you please explain what is the 1 and 4 arguments are used for ?

**Comment by erwango** [comment]
@sercantun, information is available in `dts/bindings/pwm/st,stm32-pwm.yaml`:
```
pwm-cells:
  - channel
# period in terms of nanoseconds
  - period
```

**Comment by erwango** [comment]
@giellamoswhard can you check #22277 answers your question ?

**Comment by giellamoswhard** [comment]
Yes. Thank you very much.

**Comment by erwango** [comment]
I'm reopening the point.
It will be closed automatically when #22777 gets merged

**Comment by zacck** [comment]
I find thaty the example app works on the stm32f4_disco | STM32F4_DISCOVERY. one just has to use a scope to observe the pins,  so likely some timing issue in the blink driver.  with the following overlay is what worked for me.

```
/ {
	example_sensor: example-sensor {
		compatible = "zephyr,example-sensor";
		input-gpios = <&gpiod 15 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
	};

	blink_led: blink-led {
		compatible = "blink-gpio-led";
		led-gpios = <&gpiod 14 GPIO_ACTIVE_HIGH>; 
		blink-period-ms = <1000000>;
	}; 
};

&gpiod{
	status = "okay"; 
};
```
